### PR TITLE
fix: artifacts naming issues

### LINKED ISSUE #930: [sozo] naming conflicts between Components & Systems
**Describe the bug**
A system & component using same name will create conflicts.
Generated artifacts are named [project_name]-[component_name].json or [project_name]-[system_name].json
Since only one artifact is created, the system will not be added to manifest.json

**To Reproduce**
Create a system & a component using same name.

**Additional context**
Using latest dojo version

### LINKED ISSUE #931: [sozo] Systems name & artifact naming
**Describe the bug**
I have a system named `game1`
Generatred artifact will be named [project_name]-game1.json

When migrating : 
`error: Problem preparing for migration.
Caused by:
    missing contract artifact for `game_1` contract`

**To Reproduce**
Create a system named `game1` or `gameA`
Try to `sozo migrate`

**Additional context**
Using latest dojo version

### PR DESCRIPTION
**Pull request by notV4l**
- fix #930 
- fix #931

### CODE REVIEW

File: crates/dojo-lang/src/compiler.rs
[the remark this answers is not in the document]
Comment (notV4l) [reply crates/dojo-lang/src/compiler.rs:1]: done

File: crates/dojo-lang/src/compiler.rs [hunk 1]
```diff
@@ -92,7 +93,18 @@ impl Compiler for DojoCompiler {
...
-            let file_name = format!("{target_name}-{contract_name}.json");
+            let contract_name_snake = contract_name.to_case(Case::Snake);
+
+            let is_model = match class.abi.clone() {
+                Some(abii) => abii.items.iter().any(|i| match i {
+                    abi::Item::Struct(s) => s.name == "dojo::database::schema::Struct",
+                    _ => false,
+                }),
+                None => false,
+            };
```
[the remark this answers is not in the document]
Comment (notV4l) [reply crates/dojo-lang/src/compiler.rs:104]: i believe

File: crates/dojo-lang/src/compiler.rs [hunk 2]
```diff
@@ -102,9 +102,22 @@ impl Compiler for DojoCompiler {
         for (decl, class) in zip(contracts, classes) {
             let target_name = &unit.target().name;
             let contract_name = decl.submodule_id.name(db.upcast_mut());
-            let file_name = format!("{target_name}-{contract_name}.json");
+            let contract_name_snake = contract_name.to_case(Case::Snake);
 
-            let mut file = target_dir.open_rw(file_name.clone(), "output file", ws.config())?;
+            let is_model = match &class.abi {
+                Some(abi) => abi.items.iter().any(|i| match i {
```
Comment (kariy) [inline crates/dojo-lang/src/compiler.rs:108]: Recent update to Cairo 2.3.0 seems to have made `items` private which invalidates this approach.

**Comment by kariy** [comment]
i think there'll be an issue in the mapping here as well

https://github.com/dojoengine/dojo/blob/10188401a0cf88afca30956c32e0db6e7c616b08/crates/dojo-lang/src/compiler.rs#L99

similar contracts name would simply overwrite each other

File: crates/dojo-lang/src/compiler.rs (hunk 2 above)
Comment (tarrencev) [reply crates/dojo-lang/src/compiler.rs:108]: Added a solution here: https://github.com/starkware-libs/cairo/pull/4310

**Comment by ponderingdemocritus** [comment]
this still alive?

**Pull request closed without merging**